src/Common/src/CoreLib/System/Text/ValueStringBuilder.cs

```diff
@@ -253,6 +265,19 @@ public Span<char> AppendSpan(int length)
             return _chars.Slice(origPos, length);
         }
 
+        public unsafe bool IsSameString(char* ptrStr, int start, int end)
+        {
+            while (start <= end)
+            {
+                if (ptrStr[start] != _chars[start])
+                {
+                    return false;
+                }
+                start++;
+            }
+            return true;
+        }
```

Comment:
This should not be on ValueStringBuilder.  You can do such checks outside of the type simply by getting the span from the ValueStringBuilder.